### PR TITLE
viper4linux-gui: init at unstable-2022-04-23

### PR DESCRIPTION
**Pull request by wineee**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by RashiqAzhan** [comment]
Please let this be added. I need my audio fix.

File: pkgs/applications/audio/viper4linux-gui/default.nix [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+, libpulseaudio
+, makeWrapper
+, makeDesktopItem
+, viper4linux
+, wrapQtAppsHook
+}:
+
+stdenv.mkDerivation rec {
+  pname = "viper4linux-gui";
+  version = "2.2.034f3fe";
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/viper4linux-gui/default.nix:19]: Either use the release tag or the unstable format.

File: pkgs/applications/audio/viper4linux-gui/default.nix [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  installPhase = ''
+    runHook preInstall
+    install -D V4L_Frontend $out/bin/viper-gui
+    wrapProgram $out/bin/viper-gui \
+      --prefix PATH : ${lib.makeBinPath [ viper4linux gst_all_1.gstreamer.dev ]} \
+      --prefix GST_PLUGIN_SYSTEM_PATH_1_0 : ${gst_all_1.gst-plugins-viperfx}/lib/gstreamer-1.0:${gst_all_1.gst-plugins-good}/lib/gstreamer-1.0
+    mkdir -p $out/share/icons
+    cp icons/viper.png $out/share/icons/
+    mkdir -p $out/share/applications
+    cp ${desktopItem}/share/applications/viper4linux-gui.desktop $out/share/applications/
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/viper4linux-gui/default.nix:72]: Please use copyDesktopItems.

File: pkgs/applications/audio/viper4linux/default.nix [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+  installPhase = ''
+    runHook preInstall
+
+    install -D viper -t $out/bin
+    mkdir -p $out/etc/viper4linux
+    cp -r viper4linux/* $out/etc/viper4linux
+
+    wrapProgram "$out/bin/viper" \
+      --prefix PATH : $out/bin:${lib.makeBinPath [ gst_all_1.gstreamer.dev ]}  \
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/viper4linux/default.nix:43]: Why do we need dev?

File: pkgs/applications/audio/viper4linux/default.nix [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+      --prefix GST_PLUGIN_SYSTEM_PATH_1_0 : ${gst_all_1.gst-plugins-viperfx}/lib/gstreamer-1.0:${gst_all_1.gst-plugins-good}/lib/gstreamer-1.0 \
+      --prefix LD_LIBRARY_PATH :  ${libviperfx}/lib
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/viper4linux/default.nix:45]:
```suggestion
      --prefix LD_LIBRARY_PATH : ${libviperfx}/lib
```

File: pkgs/development/libraries/gstreamer/viperfx/default.nix [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+, stdenv
+, fetchFromGitHub
+, pkg-config
+, wrapGAppsHook
+, gst_all_1
+, cmake
+}:
+stdenv.mkDerivation rec {
+  pname = "gst-plugins-viperfx";
+  version = "2.1.0.a5c1b03";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/gstreamer/viperfx/default.nix:11]: Also unstable format.

**Comment by wineee** [comment]
> How do we configure it if the config path is in $out?

We don't need to modify $out, it's a default configuration.
Configuration can be overridden in `~/.config/viper4linux/`

**Review by SuperSandro2000 — COMMENTED** [review]
otherwise should be LGTM

**Comment by wineee** [comment]
@SuperSandro2000  I adjust the code a bit，and I think it should be ready for review

libviperfx use Non-GPL licenses , User must set `allowUnfree = true` to install it, using closed source libraries should be predictable

```
4. When you redistribute the contents of this repository in any form, please bring all of the contents described here to ensure that the users or recipients specify these terms.
```
I reserve these descriptions in `$out/share/licenses`

**Comment by SuperSandro2000** [comment]
🙄 closed source libraries. Great.

**Pull request merged**